src/freelist.rs

```diff
@@ -97,4 +99,9 @@ impl<T: FreeListItem> FreeList<T> {
         item.set_next_free_id(self.first_free_index);
         self.first_free_index = Some(id);
     }
+
+    /// NB: This iterates over free items too!
```

Comment:
I'm uneasy about this since it exposes internal details - is there a way to avoid it?